### DISCUSSION TITLE
An honest opinion about the `App` vs `Pages` router

### DISCUSSION THREAD

**Discussion by RemyMachado** [vercel/next.js#59368] [App Router]
Please see https://github.com/vercel/next.js/discussions/59373 instead.

**Comment by christianjuth** [comment]
For me it's not the complexity thats driving me crazy, it's the instability. The basics like running `next dev` and deploying to Vercel are failing constantly for me since upgrading to Next.js 13, 14 and using App directory.

I'm stunned that not that many people seemed to be speaking up about the instability being introduced. Next.js + React broke radio buttons for 9 months before the issue was addressed (from v13.0.1 Oct 31, 2022 - v13.5.6 Oct 18, 2023) – see #51932 and #49499. Next.js and React Core Team are working together on this. This issue was patched by React Core Team and was probably their fault. I have a lot of respect for both teams, but I think both are to blame for moving way too fast. This is especially frustrating since Facebook often says they test React code at Facebook scale before marking it as production ready. **Next.js 13 and 14 seem anything but production ready.**

The reason I switched form Create React App to Next.js years ago was that things in Next.js just seemed to work – especially when deploying to Vercel. At the time, CRA required manual webpack configuration, you had to bring your own router, and there was no recommended way to style your app (we seem to be converging on Tailwind now). As a inexperienced React developer, it was a big headache and the upopinionated framework left me lots of room to make bad technical choices that I paid for later.

I've been following React and suspense for years, but this whole suspense + server components + streaming implementation is very disappointing and confusing.

Today I can confidently say working with Next.js 13+ is the same headache that drove me away from CRA. _Not to say CRA wasn't once important, but years ago Next.js was just a much more compelling toolchain (framework? meta framework?) for me._  I'm starting to think I no longer want to work with React long term.

I think React might be going the way of jQuery and many other tools of the past: these tools were really important in their day, they influenced the industry and the tools to come after, however, after their initial contribution they slowly became a thing of the past. Perhaps React has given everything it can, and moving forward it will live on through the influence it has had on newer frameworks.

I'm curious to hear what others think. I'm not sure the issue here is with Next.js or React as a whole, but since React Core Team and Next.js team work together closely, I'm working under the assumption that both teams are at fault. However, I would be open to trying something like Remix or some other meta framework to see if React is still the best (in the opinion of other React devs) way to build apps in 2023.

**Reply by RemyMachado** [reply]
I just realised that there is a poll category that would better fit this kind of post, would you mind resharing your thoughts on this new thread please? https://github.com/vercel/next.js/discussions/59373
